servicetalk-http-netty/src/test/java/io/servicetalk/http/api/BaseContentCodingTest.java

```diff
@@ -0,0 +1,245 @@
+        return new String(payload(b), StandardCharsets.US_ASCII);
+    }
+
+    protected enum Server {
+        DEFAULT(emptyList()),
+        GZIP_ONLY(singletonList(gzipDefault())),
+        GZIP_ID(asList(gzipDefault(), identity())),
+        GZIP_DEFLATE_ID(asList(gzipDefault(), deflateDefault(), identity())),
+        ID_GZIP(asList(identity(), gzipDefault())),
+        ID_DEFLATE(asList(identity(), deflateDefault())),
+        ID_GZIP_DEFLATE(asList(identity(), gzipDefault(), deflateDefault())),
+        DEFLATE_ONLY(singletonList(deflateDefault())),
+        DEFLATE_ID(asList(deflateDefault(), identity()));
+
+        List<ContentCodec> list;
+
+        Server(List<ContentCodec> list) {
+            this.list = list;
+        }
+    }
+
+    protected enum Client {
+        DEFAULT(emptyList()),
+        GZIP_ONLY(singletonList(gzipDefault())),
+        GZIP_ID(asList(gzipDefault(), identity())),
+        GZIP_DEFLATE_ID(asList(gzipDefault(), deflateDefault(), identity())),
+        ID_ONLY(singletonList(identity())),
+        ID_GZIP(asList(identity(), gzipDefault())),
+        ID_DEFLATE(asList(identity(), deflateDefault())),
+        ID_GZIP_DEFLATE(asList(identity(), gzipDefault(), deflateDefault())),
+        DEFLATE_ONLY(singletonList(deflateDefault())),
+        DEFLATE_ID(asList(deflateDefault(), identity()));
+
+        List<ContentCodec> list;
+
+        Client(List<ContentCodec> list) {
+            this.list = list;
+        }
+    }
+
+    protected enum Request {
+        ID(identity()),
+        GZIP(gzipDefault()),
+        DEFLATE(deflateDefault());
+
+        ContentCodec codec;
+
+        Request(ContentCodec codec) {
+            this.codec = codec;
+        }
+    }
+
+    protected enum Expectation {
+        SHOULD_PASS(true),
+        SHOULD_FAIL(false);
+
+        boolean valid;
+        Expectation(boolean valid) {
+            this.valid = valid;
+        }
+    }
+
+    protected enum Protocol {
+        H1(h1Default()),
+        H2(h2Default());
+
+        HttpProtocolConfig config;
+
+        Protocol(HttpProtocolConfig config) {
+            this.config = config;
+        }
+    }
+
+    protected static class Scenario {
+        final ContentCodec requestEncoding;
+        final List<ContentCodec> clientSupported;
+        final List<ContentCodec> serverSupported;
+        final HttpProtocolConfig protocol;
+        final boolean valid;
+        final boolean isH2;
+
+        Scenario(final ContentCodec requestEncoding,
+                 final List<ContentCodec> clientSupported, final List<ContentCodec> serverSupported,
+                 final Protocol protocol, final boolean valid) {
+            this.requestEncoding = requestEncoding;
+            this.clientSupported = clientSupported;
+            this.serverSupported = serverSupported;
+            this.valid = valid;
+            this.isH2 = protocol == H2;
+            this.protocol = isH2 ? h2Default() : h1Default();
```

Comment:
Consider just storing a enum, it has all the info for `isH2`, the `protocol`, and its name for logging